--- srcOrig/fileops.c	2012-02-19 13:54:57.000000000 +0100
+++ src/fileops.c	2012-03-19 15:05:46.000000000 +0100
@@ -664,7 +664,7 @@
 	/* vi is set as the default for any extension without a program */
 	if(program == NULL)
 	{
-		if(view->dir_entry[view->list_pos].type == DIRECTORY)
+		if(view->dir_entry[view->list_pos].type == DIRECTORY || (view->dir_entry[view->list_pos].type  == LINK && is_dir(view->dir_entry[view->list_pos].name)))
 		{
 			handle_dir(view);
 		}
@@ -908,7 +908,7 @@
 	if(cfg.vim_filter && (executable || runnable))
 		use_vim_plugin(view, 0, NULL); /* no return */
 
-	if(executable && type != DIRECTORY)
+	if(executable && type != DIRECTORY&&(type != LINK || !is_dir(filename)))
 	{
 		char buf[NAME_MAX];
 		snprintf(buf, sizeof(buf), "%s/%s", view->curr_dir, filename);
