--- srcOrig/filelist.c	2012-03-21 09:47:13.000000000 +0100
+++ src/filelist.c	2012-03-21 09:25:54.000000000 +0100
@@ -1985,8 +1985,8 @@
 		size_t name_len;
 		struct stat s;
 
-		/* Ignore the "." directory. */
-		if(pathcmp(d->d_name, ".") == 0)
+		/* Ignore the "." directory unless it is a root directory. */
+		if((pathcmp(d->d_name, ".") == 0)&&!is_root_dir(view->curr_dir))
 		{
 			view->list_rows--;
 			continue;
@@ -2134,8 +2134,8 @@
 		dir_entry_t *dir_entry;
 		size_t name_len;
 
-		/* Ignore the "." directory. */
-		if(pathcmp(ffd.cFileName, ".") == 0)
+		/* Ignore the "." directory unless it is the root directory*/
+		if((pathcmp(ffd.cFileName, ".") == 0)&&!is_root_dir(view->curr_dir))
 			continue;
 		/* Always include the ../ directory unless it is the root directory. */
 		if(pathcmp(ffd.cFileName, "..") == 0)
