diff -rupN okular-4.11.5/ui/pageview.cpp okular-4.11.5-New/ui/pageview.cpp
--- okular-4.11.5/ui/pageview.cpp	2014-01-02 20:33:09.000000000 +0100
+++ okular-4.11.5-New/ui/pageview.cpp	2014-02-07 17:47:32.661288618 +0100
@@ -1637,8 +1637,8 @@ void PageView::keyPressEvent( QKeyEvent
     // move/scroll page by using keys
     switch ( e->key() )
     {
-        case Qt::Key_J:
-        case Qt::Key_K:
+        case Qt::Key_T:
+        case Qt::Key_C:
         case Qt::Key_Down:
         case Qt::Key_PageDown:
         case Qt::Key_Up:
@@ -1646,14 +1646,14 @@ void PageView::keyPressEvent( QKeyEvent
         case Qt::Key_Backspace:
             if ( e->key() == Qt::Key_Down
                  || e->key() == Qt::Key_PageDown
-                 || e->key() == Qt::Key_J )
+                 || e->key() == Qt::Key_T )
             {
-                bool singleStep = e->key() == Qt::Key_Down || e->key() == Qt::Key_J;
+                bool singleStep = e->key() == Qt::Key_Down || e->key() == Qt::Key_T;
                 slotScrollDown( singleStep );
             }
             else
             {
-                bool singleStep = e->key() == Qt::Key_Up || e->key() == Qt::Key_K;
+                bool singleStep = e->key() == Qt::Key_Up || e->key() == Qt::Key_C;
                 slotScrollUp( singleStep );
             }
             break;
@@ -1669,6 +1669,7 @@ void PageView::keyPressEvent( QKeyEvent
                 horizontalScrollBar()->triggerAction( QScrollBar::SliderSingleStepSub );
             break;
         case Qt::Key_Right:
+        case Qt::Key_N:
         case Qt::Key_L:
             if ( horizontalScrollBar()->maximum() == 0 )
             {
