--- pageview.cpp	2012-04-24 17:04:00.000000000 +0200
+++ page.cppNew	2012-04-24 17:03:52.000000000 +0200
@@ -1283,8 +1283,8 @@ void PageView::keyPressEvent( QKeyEvent
     // move/scroll page by using keys
     switch ( e->key() )
     {
-        case Qt::Key_J:
-        case Qt::Key_K:
+        case Qt::Key_T:
+        case Qt::Key_C:
         case Qt::Key_Down:
         case Qt::Key_PageDown:
         case Qt::Key_Space:
@@ -1293,13 +1293,13 @@ void PageView::keyPressEvent( QKeyEvent
         case Qt::Key_Backspace:
             if ( e->key() == Qt::Key_Down
                  || e->key() == Qt::Key_PageDown
-                 || e->key() == Qt::Key_J
+                 || e->key() == Qt::Key_T
                  || ( e->key() == Qt::Key_Space && ( e->modifiers() & Qt::ShiftModifier ) != Qt::ShiftModifier ) )
             {
                 // if in single page mode and at the bottom of the screen, go to next page
                 if ( Okular::Settings::viewContinuous() || verticalScrollBar()->value() < verticalScrollBar()->maximum() )
                 {
-                    if ( e->key() == Qt::Key_Down || e->key() == Qt::Key_J )
+                    if ( e->key() == Qt::Key_Down || e->key() == Qt::Key_T )
                         verticalScrollBar()->triggerAction( QScrollBar::SliderSingleStepAdd );
                     else
                         verticalScrollBar()->triggerAction( QScrollBar::SliderPageStepAdd );
@@ -1321,7 +1321,7 @@ void PageView::keyPressEvent( QKeyEvent
                 // if in single page mode and at the top of the screen, go to \ page
                 if ( Okular::Settings::viewContinuous() || verticalScrollBar()->value() > verticalScrollBar()->minimum() )
                 {
-                   if ( e->key() == Qt::Key_Up || e->key() == Qt::Key_K )
+                   if ( e->key() == Qt::Key_Up || e->key() == Qt::Key_C )
                         verticalScrollBar()->triggerAction( QScrollBar::SliderSingleStepSub );
                    else
                        verticalScrollBar()->triggerAction( QScrollBar::SliderPageStepSub );
@@ -1344,7 +1344,7 @@ void PageView::keyPressEvent( QKeyEvent
             horizontalScrollBar()->triggerAction( QScrollBar::SliderSingleStepSub );
             break;
         case Qt::Key_Right:
-        case Qt::Key_L:
+        case Qt::Key_N:
             horizontalScrollBar()->triggerAction( QScrollBar::SliderSingleStepAdd );
             break;
         case Qt::Key_Escape:
